opent2t.p.light/com.wink.lightbulb/js/thingTranslator.js

```diff
@@ -0,0 +1,243 @@
+// we use this StateReader class to read from desired_state if it is there, and fall back to last_reading
+// if it is not.
+class StateReader {
+    constructor(desired_state, last_reading) {
+        this.desired_state = desired_state;
+        this.last_reading = last_reading;
+    }
+
+    get(state) {
+        if (this.desired_state[state] !== undefined) {
+            return this.desired_state[state];
+        }
+        else {
+            return this.last_reading[state];
+        }
+    }
+}
+
+// Helper method to convert Wink's 0.0-1.0 brightness scale to a 0-100 scale
+function scaleDeviceBrightnessToTranslatorBrightness(brightnessValue) {
+    return Math.floor(brightnessValue * 100);
+}
+
+// Helper method to convert a 0-100 scale to Wink's 0.0-1.0 brightness scale
+function scaleTranslatorBrightnessToDeviceBrightness(dimmingValue) {
+    return dimmingValue / 100;
+}
+
+/**
+ * Finds a resource for an entity in a schema
+ */
+function findResource(schema, di, resourceId) {
+    // Find the entity by the unique di
+    var entity = schema.entities.find((d) => {
+        return d.di === di;
+    });
+
+    // Find the resource
+    var resource = entity.resources.find((r) => {
+        return r.id === resourceId;
+    });
+
+    return resource;
+}
+
+/***
+ * 
+ */
+function resourceSchemaToProviderSchema(resourceId, resourceSchema) {
+    // build the object with desired state
+    var result = { 'desired_state': {} };
+    var desired_state = result.desired_state;
+
+    switch(resourceId) {
+        case 'power':
+            desired_state['powered'] = resourceSchema.value;
+            break;
+        case 'dim':
+            desired_state['brightness'] = scaleTranslatorBrightnessToDeviceBrightness(resourceSchema.dimmingSetting);
+            break;
+        default:
+            // Error case
+            throw new Error("Invalid resourceId");
+    }
+
+    return result;
+}
+
+/**
+ * 
+ */
+function providerSchemaToPlatformSchema(providerSchema, expand) {
+    var stateReader = new StateReader(providerSchema.desired_state, providerSchema.last_reading);
+
+    // Build the oic.r.switch.binary resource
+    var power = {
+        "href": "/power",
+        "rt": ["oic.r.switch.binary"],
+        "if": ["oic.if.a", "oic.if.baseline"]
+    }
+
+    // Build the oic.r.dimming resource
+    var dim = {
+         "href": "/dim",
+         "rt": ["oic.r.light.dimming"],
+         "if": ["oic.if.a", "oic.if.baseline"]
+    }
+
+    // Include the values is expand is specified
+    if (expand) {
+        power.id = 'power';
+        power.value = stateReader.get('powered');
+
+        dim.id = 'dim';
+        dim.dimmingSetting = scaleDeviceBrightnessToTranslatorBrightness(stateReader.get('brightness'));
+        dim.range = [0,100];
+    }
+
+    return {
+        opent2t: {
+            schema: 'opent2t.p.light',
+            translator: 'opent2t-translator-com-wink-lightbulb',
+            control_id: providerSchema['object_id']
```

Comment:
I think we've been using camelCase instead of underscores for other JSON field names, we probably want to stick with that, so controlId instead of control_id.